extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,461 @@
+            }
+            return null;
+        }
+
+        @Override
+        public String target(final HttpRequest request) {
+            return request.uri();
+        }
+
+        @Override
+        public String route(final HttpRequest request) {
+            return request.uri().length() > 1 ? request.uri() : null;
+        }
+
+        @Override
+        public String scheme(final HttpRequest request) {
+            if (request instanceof HttpServerRequest) {
+                return ((HttpServerRequest) request).scheme();
+            }
+            return null;
+        }
+
+        @Override
+        public String serverName(final HttpRequest request, final HttpResponse response) {
+            return request.remoteAddress().hostName();
+        }
+
+        @Override
+        public String method(final HttpRequest request) {
+            return request.method().name();
+        }
+
+        @Override
+        public List<String> requestHeader(final HttpRequest request, final String name) {
+            return request.headers().getAll(name);
+        }
+
+        @Override
+        public Long requestContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long requestContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Integer statusCode(final HttpRequest request, final HttpResponse response) {
+            return response.statusCode();
+        }
+
+        @Override
+        public Long responseContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long responseContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public List<String> responseHeader(final HttpRequest request, final HttpResponse response, final String name) {
+            return response.headers().getAll(name);
+        }
+    }
+
+    // TODO - Ideally this should use HttpSpanNameExtractor, but to keep the name without the slash we use our own.
+    private static class ServerSpanNameExtractor implements SpanNameExtractor<HttpRequest> {
+        private final HttpServerAttributesGetter<HttpRequest, HttpResponse> serverAttributesExtractor;
+
+        ServerSpanNameExtractor(
+                final HttpServerAttributesGetter<HttpRequest, HttpResponse> serverAttributesExtractor) {
+            this.serverAttributesExtractor = serverAttributesExtractor;
+        }
+
+        @Override
+        public String extract(final HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpServerRequest) {
+                String path = URI.create(httpRequest.uri()).getPath();
+                if (path != null && path.length() > 1) {
+                    return path.substring(1);
```

Comment:
Why does one of the branches uses HTTP + verb, while here we just have the destination? Shouldn't it be consistent?

Reply:
It is how OTel Semantic Convention defined it:

> HTTP spans MUST follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span). Many REST APIs encode parameters into URI path, e.g. /api/users/123 where 123 is a user id, which creates high cardinality value space not suitable for span names. In case of HTTP servers, these endpoints are often mapped by the server frameworks to more concise HTTP routes, e.g. /api/users/{user_id}, which are recommended as the low cardinality span names. However, the same approach usually does not work for HTTP client spans, especially when instrumentation is provided by a lower-level middleware that is not aware of the specifics of how the URIs are formed. Therefore, HTTP client spans SHOULD be using conservative, low cardinality names formed from the available parameters of an HTTP request, such as "HTTP {METHOD_NAME}". Instrumentation MUST NOT default to using URI path as span name, but MAY provide hooks to allow custom logic to override the default span name.

And that is what they do in their implementation:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java#L34-L45

Reply:
ah ok... Thanks!